utilities/tools/Test-NamePrefixAvailability.ps1

```diff
@@ -56,127 +64,164 @@ function Test-NamePrefixAvailability {
-            foreach ($storageAccountName in $storageAccountNames) {
-                $path = '/subscriptions/{0}/providers/Microsoft.Storage/checkNameAvailability?api-version=2021-04-01' -f $subscriptionId
-                $requestInputObject = @{
-                    Method  = 'POST'
-                    Path    = $path
-                    Payload = @{
-                        name = $storageAccountName
-                        type = 'Microsoft.Storage/storageAccounts'
-                    } | ConvertTo-Json
+                # determine serviceshort default value if no parameter has been supplied
+                if (!$overwrittenServiceShort) {
+                    $serviceShort = Get-Content -Path $parameterFile | ForEach-Object {
+                        if ($_ -match "serviceShort string = ") { $_ }
+                    }
+                    $serviceShort = $serviceShort.Split("=")[-1] # split string to get the default value for serviceshort
+                    $serviceShort = $serviceShort.Replace("'", '') # remove trailing quotes
+                    $serviceShort = $serviceShort.Replace("'", '') # remove trailing quotes
                 }
-                $response = ((Invoke-AzRestMethod @requestInputObject).Content | ConvertFrom-Json)
-                if ($response.nameAvailable) { Write-Host "=> The storage account name [$storageAccountName] is currently available" -ForegroundColor 'Green' }
                 else {
-                    Write-Warning "=> Warning: The storage account name [$storageAccountName] is not available. Please try a different prefix."
-                    $prefixAvailable = $false
+                    $serviceShort = $overwrittenServiceShort
                 }
-            }
 
-            # Key Vault
-            Write-Host "`nChecking Key Vaults" -ForegroundColor 'Cyan'
-            Write-Host '===================' -ForegroundColor 'Cyan'
-            foreach ($keyVaultName in $keyVaultNames) {
-                $path = '/subscriptions/{0}/providers/Microsoft.KeyVault/checkNameAvailability?api-version=2021-10-01' -f $subscriptionId
-                $requestInputObject = @{
-                    Method  = 'POST'
-                    Path    = $path
-                    Payload = @{
-                        name = $keyVaultName
-                        type = 'Microsoft.KeyVault/vaults'
-                    } | ConvertTo-Json
-                }
-                $response = ((Invoke-AzRestMethod @requestInputObject).Content | ConvertFrom-Json)
-                if ($response.nameAvailable) { Write-Host "=> The key vault name [$keyVaultName] is currently available" -ForegroundColor 'Green' }
-                else {
-                    Write-Warning "=> Warning: The key vault name [$keyVaultName] is not available. Please try a different prefix."
-                    $prefixAvailable = $false
+                if ($temp) {
+                    $temp = $temp.Split($filter)
```

Comment:
Also this could be solved via regex & group matchings if you're up to the task ;) would probably save you 10+ lines

Reply:
Let's save that for later to get the PR in